modules/phase_field/src/ics/MultiSmoothSuperellipsoidIC.C

```diff
@@ -95,29 +95,48 @@ MultiSmoothSuperellipsoidIC::computeSuperellipsoidSemiaxes()
 {
   for (unsigned int i = 0; i < _numbub[_gk]; i++)
   {
-    //Vary bubble radius
     switch (_semiaxis_variation_type)
     {
-    case 0: //Uniform distrubtion
-      _as.push_back(_semiaxis_a[_gk]*(1.0 + (1.0 - 2.0 * _random.rand(_tid)) * _semiaxis_a_variation[_gk]));
-      _bs.push_back(_semiaxis_b[_gk]*(1.0 + (1.0 - 2.0 * _random.rand(_tid)) * _semiaxis_b_variation[_gk]));
-      _cs.push_back(_semiaxis_c[_gk]*(1.0 + (1.0 - 2.0 * _random.rand(_tid)) * _semiaxis_c_variation[_gk]));
-      break;
-    case 1: //Normal distribution
-      _as.push_back(_random.randNormal(_tid, _semiaxis_a[_gk],_semiaxis_a_variation[_gk]));
-      _bs.push_back(_random.randNormal(_tid, _semiaxis_b[_gk],_semiaxis_b_variation[_gk]));
-      _cs.push_back(_random.randNormal(_tid, _semiaxis_c[_gk],_semiaxis_c_variation[_gk]));
-      break;
-    case 2: //No variation
-      _as.push_back(_semiaxis_a[_gk]);
-      _bs.push_back(_semiaxis_b[_gk]);
-      _cs.push_back(_semiaxis_c[_gk]);
-    }
+      case 0: //Uniform distrubtion
+        if (_vary_axes_independently)
+        {
+          _as.push_back(_semiaxis_a[_gk] * (1.0 + (1.0 - 2.0 * _random.rand(_tid)) * _semiaxis_a_variation[_gk]));
+          _bs.push_back(_semiaxis_b[_gk] * (1.0 + (1.0 - 2.0 * _random.rand(_tid)) * _semiaxis_b_variation[_gk]));
+          _cs.push_back(_semiaxis_c[_gk] * (1.0 + (1.0 - 2.0 * _random.rand(_tid)) * _semiaxis_c_variation[_gk]));
+        }
+        else
+        {
+          const Real randnum = _random.rand(_tid);
+          _as.push_back(_semiaxis_a[_gk] * (1.0 + (1.0 - 2.0 * randnum) * _semiaxis_a_variation[_gk]));
+          _bs.push_back(_semiaxis_b[_gk] * (1.0 + (1.0 - 2.0 * randnum) * _semiaxis_b_variation[_gk]));
+          _cs.push_back(_semiaxis_c[_gk] * (1.0 + (1.0 - 2.0 * randnum) * _semiaxis_c_variation[_gk]));
+        }
+        break;
 
-    if (_as.back() < 0.0) _as.back() = 0.0;
-    if (_bs.back() < 0.0) _bs.back() = 0.0;
-    if (_cs.back() < 0.0) _cs.back() = 0.0;
+      case 1: //Normal distribution
+        if (_vary_axes_independently)
+        {
+          _as.push_back(_random.randNormal(_tid, _semiaxis_a[_gk],_semiaxis_a_variation[_gk]));
+          _bs.push_back(_random.randNormal(_tid, _semiaxis_b[_gk],_semiaxis_b_variation[_gk]));
+          _cs.push_back(_random.randNormal(_tid, _semiaxis_c[_gk],_semiaxis_c_variation[_gk]));
+        }
+        else
+        {
+          const Real randnum = _random.randNormal(_tid, 0, 1);
+          _as.push_back(_semiaxis_a[_gk] + (randnum * _semiaxis_a_variation[_gk]));
+          _bs.push_back(_semiaxis_b[_gk] + (randnum * _semiaxis_b_variation[_gk]));
+          _cs.push_back(_semiaxis_c[_gk] + (randnum * _semiaxis_c_variation[_gk]));
+        }
+        break;
 
+      case 2: //No variation
+        _as.push_back(_semiaxis_a[_gk]);
+        _bs.push_back(_semiaxis_b[_gk]);
+        _cs.push_back(_semiaxis_c[_gk]);
```

Comment:
I use this IC to create multiple sets of superellipsoids, each set with its own semiaxis, axis variation, number of bubble etc, but i still need to maintain the full list of parameters across each set so that i can check them against each other for overlap.